File: src/main/java/com/example/demoimdb/model/MovieDirector.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "movie_director")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class MovieDirector {
    @ManyToOne
    @JoinColumn(name = "director_id")
    private Director director;
    @ManyToOne
    @JoinColumn(name = "movie_id")
    @JsonIgnore
    private Movie movie;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
}

File: src/entities/MovieDirector.ts
import { Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Director } from './Director';
import { Movie } from './Movie';

@Entity({ name: 'movie_director' })
export class MovieDirector {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(() => Movie, (movie) => movie.movieDirectors, {
    onDelete: 'SET NULL',
  })
  movie: Movie;

  @ManyToOne(() => Director, (director) => director.movieDirectors)
  director: Director;
}
